Mathlib/Data/Matrix/Basic.lean

```diff
@@ -1211,6 +1212,63 @@ end Algebra
 
 end Matrix
 
+section apply₂
+
+namespace AddMonoidHom
+
+variable [AddZeroClass α]
+
+variable (R α) in
+/--
+Extracting entries from a matrix as a monoid homomorphism. Note this cannot be upgraded to a ring
+homomorphism, as it does not respect multiplication.
+-/
+@[simps]
+def apply₂ (i : m) (j : n) : Matrix m n α →+ α where
```

Comment:
Maybe `Matrix.elemAddMonoidHom` and `Matrix.elimLinearMap` as the names?

Reply:
I don't mind `elem`, but the pattern for `mapMatrix` is to prefix with the structure, eg `LinearMap.mapMatrix` and `AddMonoidHom.mapMatrix`, so I'd prefer to stick with that.

Reply:
Those are named that way for `f.mapMatrix M` dot notation, since `M.mapRingHom f` doesn't work with morphisms and is morally the wrong order for function application. I don't think that applies here.

Reply:
Either way, I think this should include the word `Matrix` in the name somewhere, namespace or otherwise.

Reply:
Ah fair point - I didn't consider the dot notation. Will amend.

Reply:
done